### ISSUE TITLE
Should conditional type evaluation destroy data?

### ISSUE THREAD

**Issue by Anthropic** [Textalk/angular-schema-form#204]
I have an attribute directive I applied to my own decorator to remove data that was in elements hidden by a conditional type evaluation.

Should this perhaps be included in the main angular-schema-form behaviour?

```javascript
(function(window, angular, undefined) {'use strict';
  angular
    .module('schemaForm.extension',[])
    .directive('destroyHiddenData', ['oySchemaFormUtil', function(oySchemaFormUtil) {
      return {
        link: function(scope, element, attrs) {
          scope.$on('$destroy', function(){
            if(scope.form.key) oySchemaFormUtil.setKeyValue(scope.model, scope.form.key, '');
          })
        }
      };
    }]);
})(window, window.angular);
```
Note: the oySchemaFormUtil is my own function that just manipulates data based on key, but unlike the built in one it also allows for total deletion which is why I use it instead. Not relevant to discuss in this post thread.

**Comment by davidlgj** [comment]
Yes, this might be a good idea... It would make conditional even more useful.

Maybe an option to turn it off if needed?

**Comment by Anthropic** [comment]
I would suggest that the option allow for setting the value to a user preferred ( null, undefined, empty string or the current behaviour of do nothing) value, as in many cases it completely depends on how they then save the data to their database.

**Comment by Anthropic** [comment]
Upon reflection I would like the default behaviour to set the value to null. It should not delete it as that is not ideal in JavaScript and the back end will need to be able to see changes without relying on every field always having a value.

**Comment by jbsaff** [comment]
@Anthropic  I took a stab at this feature in https://github.com/Textalk/angular-schema-form/issues/371. I opted to use undefined instead of null, as that would most closely represent the state of how the user experienced the form (they don't see the question, so it never should come up in the model). However, I did include a check for a formDefaults definition of the destroyStrategy, which would allow for the entire form to use null as the mechanism for destroying data that has been hidden.
